src/main/java/org/myrobotlab/service/interfaces/ConnectionEventListener.java

```diff
@@ -0,0 +1,23 @@
+package org.myrobotlab.service.interfaces;
+
+import okhttp3.Response;
+import okhttp3.WebSocket;
+
+/**
+ * Start with interface exported from okhttpe. Next abstract it
+ * and use only pojos for parameters. 
+ * 
+ * TODO - use for other byte stream event handlers like serial port listeners
+ * 
+ * @author GroG
+ *
+ */
+public interface ConnectionEventListener {
+
+  public void onOpen(WebSocket webSocket, Response response);
+  
+  public void onClosing(WebSocket webSocket, int code, String reason);
```

Comment:
why not onClosed ?  this lifecycle happens just before the socket is closed perhaps?

Reply:
it has both ...
```java
  /**
   * Invoked when the remote peer has indicated that no more incoming messages will be
   * transmitted.
   */
  public void onClosing(WebSocket webSocket, int code, String reason) {
  }

  /**
   * Invoked when both peers have indicated that no more messages will be transmitted and the
   * connection has been successfully released. No further calls to this listener will be made.
   */
  public void onClosed(WebSocket webSocket, int code, String reason) {
  }
```